ui/src/app/views/profile-page/component.ts

```diff
@@ -47,16 +71,62 @@ export class ProfilePageComponent implements OnInit {
       });
   }
 
+  public get completedTasks() {
+    if (this.profile === undefined) {
+      return 0;
```

Comment:
You can put the initializer above this and `return completedTasks;` here - removes another magic number (not _very_ magic b/c the definition is right below, but still).